examples/deployments/2.clusters/2.prometheus-output/kubernetes/gnmic-app/statefulset.yaml

```diff
@@ -0,0 +1,70 @@
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
Hi @melkypie 
Great PR, thanks.
Just for my understanding, what are the benefits of a `ss` here vs `dep`?

Reply:
The leader of the gnmic cluster needs to be able to reach other cluster instances to assign them targets. If we used a deployment that is not possible. We can only reach other instances in a `Deployment` by using a `Service` which would randomly select a pod that is assigned to said service (this is also the reason why we need to use a headless `Service` ( service with `clusterIP: none`) for Prometheus service discovery as the DNS will return all `StatefulSet` pod IPs rather than just one of them). With a `StatefulSet` each of the pods gets it's own specific DNS name, e.g. `gnmic-ss-0.gnmic-svc.gnmic.cluster.local`, that allows the leader to contact each pod individually.